src/test/java/com/karankumar/bookproject/backend/service/AuthorServiceTest.java

```diff
@@ -0,0 +1,81 @@
+package com.karankumar.bookproject.backend.service;
+
+import com.karankumar.bookproject.annotations.IntegrationTest;
+import com.karankumar.bookproject.backend.entity.Author;
+import com.karankumar.bookproject.backend.repository.AuthorRepository;
+import lombok.extern.java.Log;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@Log
+@IntegrationTest
+public class AuthorServiceTest {
+
+    private final AuthorService sut;
+    private final BookService bookService;
+    private final AuthorRepository authorRepository;
+
+    @Autowired
+    public AuthorServiceTest(AuthorService sut, BookService bookService, AuthorRepository authorRepository) {
+
+        this.sut = sut;
+        this.bookService = bookService;
+        this.authorRepository = authorRepository;
+    }
+
+    @BeforeEach
+    void teardown() {
+        bookService.deleteAll();
+        authorRepository.deleteAll();
+    }
+
+    @Test
+    public void testSaveAndConfirmDuplicateNameWithDifferentId() {
+
+        //given
+        Author author = new Author("Nyor", "Ja");
+        sut.save(author);
+
+        Author author1 = author;
+        sut.save(author1);
+
+        Author author2 = new Author("Papa", "Cologne");
+        sut.save(author2);
+        sut.save(author2);
+
+        // when
```

Comment:
In this `when` section, I think it would be better to have the `save()` methods here, so that we have:

- Given an author and another author with the same author
- When we try to save authors with the same name
- We get back [expected]